src/components/article-page/article-page.jsx

```diff
@@ -45,13 +50,38 @@ function ArticlePage() {
     try {
       const incremenmt = voted ? -1 : 1;
       setVotes((prevVotes) => prevVotes + incremenmt);
-      setVoted(prevVoted => !prevVoted);
+      setVoted((prevVoted) => !prevVoted);
       await patchVoteByArticleID(article_id, { inc_votes: incremenmt });
     } catch (err) {
       setVotes((prevVotes) => prevVotes - incremenmt);
-      setVoted(prevVoted => !prevVoted);
+      setVoted((prevVoted) => !prevVoted);
+    }
+  }
+
+  async function handleSubmit(event) {
+    event.preventDefault();
+    try {
+      setCommentsArray([
+        {
+          body: comment,
+          author: currUser.username,
+          votes: 0,
+          created_at: new Date(),
+        },
+        ...commentsArray,
+      ]);
+      console.log(`before post`);
+      const x = await postCommentByArticleID(currArticle.article_id, {
+        body: comment,
+        username: currUser.username,
+      });
+      console.log(x);
+      setComment("");
+    } catch (err) {
+      console.log(err);
```

Comment:
remember to undo the optimistic render if the post request fails - (we want to remove it from the comments array)